src/test/java/uk/gov/ons/census/casesvc/service/QuestionnaireLinkedServiceTest.java

```diff
@@ -102,58 +90,46 @@ public void testGoodQuestionnaireLinkedForReceiptedCase() {
     testUacQidLink.setActive(false);
     testUacQidLink.setCaze(null);
 
-    when(uacQidLinkRepository.findByQid(TEST_QID)).thenReturn(Optional.of(testUacQidLink));
+    when(uacService.findByQid(TEST_QID)).thenReturn(testUacQidLink);
     when(caseService.getCaseByCaseId(TEST_CASE_ID)).thenReturn(testCase);
 
     // WHEN
     underTest.processQuestionnaireLinked(managementEvent);
 
     // THEN
     ArgumentCaptor<Case> caseCaptor = ArgumentCaptor.forClass(Case.class);
-    verify(caseRepository).saveAndFlush(caseCaptor.capture());
+    verify(caseService).saveAndEmitCaseUpdatedEvent(caseCaptor.capture());
     Case actualCase = caseCaptor.getValue();
     assertThat(actualCase.getCaseId()).isEqualTo(TEST_CASE_ID);
     assertThat(actualCase.isReceiptReceived()).isTrue();
 
-    verify(caseService).emitCaseUpdatedEvent(testCase);
-
     ArgumentCaptor<UacQidLink> uacQidLinkCaptor = ArgumentCaptor.forClass(UacQidLink.class);
-    verify(uacQidLinkRepository).saveAndFlush(uacQidLinkCaptor.capture());
+    verify(uacService).saveAndEmitUacUpdatedEvent(uacQidLinkCaptor.capture());
     UacQidLink actualUacQidLink = uacQidLinkCaptor.getValue();
     assertThat(actualUacQidLink.getCaze()).isEqualTo(testCase);
     assertThat(actualUacQidLink.isActive()).isFalse();
-
-    verify(uacService).emitUacUpdatedEvent(testUacQidLink, testCase);
     verify(eventLogger)
         .logUacQidEvent(
             eq(testUacQidLink),
             any(OffsetDateTime.class),
-            any(OffsetDateTime.class),
             eq("Questionnaire Linked"),
             eq(EventType.QUESTIONNAIRE_LINKED),
             eq(managementEvent.getEvent()),
             anyString());
-
-    verifyNoMoreInteractions(caseRepository);
   }
 
   @Test(expected = RuntimeException.class)
   public void testQuestionnaireLinkedQidNotFound() {
     // GIVEN
     ResponseManagementEvent managementEvent = getTestResponseManagementQuestionnaireLinkedEvent();
     String questionnaireId = managementEvent.getPayload().getUac().getQuestionnaireId();
-    String expectedErrorMessage =
-        String.format("Questionnaire Id '%s' not found!", questionnaireId);
-
-    when(uacQidLinkRepository.findByQid(questionnaireId)).thenReturn(Optional.empty());
-
-    try {
-      // WHEN
-      underTest.processQuestionnaireLinked(managementEvent);
-    } catch (RuntimeException re) {
-      // THEN
-      assertThat(re.getMessage()).isEqualTo(expectedErrorMessage);
-      throw re;
-    }
+
```

Comment:
does this test still belong here?  A similar test is missing from UacServiceTest